middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -26,6 +26,25 @@ module C = struct
   include To_cmm_shared
 end
 
+let sign_extend_63 dbg e =
+  let open Cmm in
+  Cop
+    ( Casr,
+      [Cop (Clsl, [e; Cconst_int (1, dbg)], dbg); Cconst_int (1, dbg)],
```

Comment:
here, you can call `low_63` on `e` before sign extending

Reply:
I basically copied an adapted `Cmm_helpers.sign_extend_32`,
which does not contain any calls to `low_32`. Why would this one
be different?

Reply:
I don't think I agree, there is a call to `low_32` in `sign_extend_32`.  I'm adding it to `sign_extend_63`.